### PR TITLE
Handle known opengraph namespaces

### PR DESCRIPTION
**Pull request by jakubwasikowski**
Added the following things for OpenGraph:
- Defined and handled known OpenGraph namespaces.
- Added compiling regex.
- Add extraction of prefixes from html element

### CODE REVIEW

File: extruct/opengraph.py [hunk 1]
```diff
@@ -2,6 +2,17 @@
 import lxml.html
 
 
+_PREFIX_PATTERN = re.compile(r'\s*(\w+): ([^\s]+)')
```
Comment (kmike) [inline extruct/opengraph.py]: I wonder if we should be less strict regarding whitespaces after `:`; in http://ogp.me/ I found nothing about a single whitespace being required. Something like `r'\s*(\w+):\s*([^\s]+)'`?

File: extruct/opengraph.py [hunk 2]
```diff
@@ -12,16 +23,24 @@ def extract(self, htmlstring, base_url=None, encoding='UTF-8'):
 
     def extract_items(self, document, base_url=None):
         # OpenGraph defines a web page as a single rich object.
-        # TODO: Handle known opengraph namespaces.
         for head in document.xpath('//head'):
-            prefix = dict(re.findall(r'\s*(\w+): ([^\s]+)', head.attrib.get('prefix', '')))
-            prefix.setdefault('og', 'http://ogp.me/ns#')
+            html_elems = document.head.xpath("parent::html")
```
Comment (kmike) [inline extruct/opengraph.py]: 👍

**Comment by kmike** [comment]
Looks great, thanks @jakubwasikowski! 

There is a "Roadmap" section in README - do you think it can be removed now?

I'm not sure I've read https://github.com/scrapinghub/extruct/issues/31 correctly, but it could also be related.

File: tests/samples/misc/opengraph_flat_test.json [hunk 3]
```diff
@@ -5,9 +5,14 @@
       "og:image": "https://www.eventeducation.com/images/982336_wedding_dayandouan_th.jpg",
       "og:site_name": "Event Education",
       "og:description": "Event Education provides free courses on event planning and management to event professionals worldwide.",
+      "article:publisher": "http://www.facebook.com/PUBLISHER",
```
Comment (lopuhin) [inline tests/samples/misc/opengraph_flat_test.json]: I wonder what was result of parsing ``<meta property="article:publisher" content="http://www.facebook.com/PUBLISHER" />`` before this PR?

File: extruct/opengraph.py (hunk 1 above)
Comment (jakubwasikowski) [reply extruct/opengraph.py]: I've tried to find if there is any specification about that, but didn't find any. Agree, will be better to put `\s*` there :+1:

File: tests/samples/misc/opengraph_flat_test.json (hunk 3 above)
Comment (jakubwasikowski) [reply tests/samples/misc/opengraph_flat_test.json]: Here is the result of parsing before that PR:
```
{
    "@context": {
        "og": "http://ogp.me/ns#"
    },
    "@type": "article",
    "og:description": "Event Education provides free courses on event planning and management to event professionals worldwide.",
    "og:image": "https://www.eventeducation.com/images/982336_wedding_dayandouan_th.jpg",
    "og:site_name": "Event Education",
    "og:title": "Himanshu's Open Graph Protocol",
    "og:url": "https://www.eventeducation.com/test.php"
}
```
And after:
```
{
    "@context": {
        "article": "http://ogp.me/ns/article#",
        "og": "http://ogp.me/ns#"
    },
    "@type": "article",
    "article:author": "http://facebook.com/AUTHOR",
    "article:modified_time": "2016-12-13T12:35:52-07:00",
    "article:published_time": "2012-04-04T19:50:00-07:00",
    "article:publisher": "http://www.facebook.com/PUBLISHER",
    "og:description": "Event Education provides free courses on event planning and management to event professionals worldwide.",
    "og:image": "https://www.eventeducation.com/images/982336_wedding_dayandouan_th.jpg",
    "og:site_name": "Event Education",
    "og:title": "Himanshu's Open Graph Protocol",
    "og:url": "https://www.eventeducation.com/test.php"
}
```
Property `article:publisher` would be in the output for the version before only if there is prefix for articles defined in `<head>` tag, but for that test case it doesn't take place. Furthermore I noticed in our test case that this is quite rare situation.

**Comment by jakubwasikowski** [comment]
> I'm not sure I've read #31 correctly, but it could also be related.

@kmike, I'm not sure what @redapple exactly meant by "while they could use the type-dependent OGP namespace", but I think this is suggestion that to implement interpretation of value of property (for instance casting to specified type) depending on the type of schema from `og:type`.

I'm not doing that in the PR - I've just defined that some schemas are known and "pre-defined" in OGP and there is no need to specify these schemas in `prefix` attribute. I'm not interpreting here values of these properties.

File: tests/samples/misc/opengraph_flat_test.json (hunk 3 above)
Comment (lopuhin) [reply tests/samples/misc/opengraph_flat_test.json]: @jakubwasikowski thanks for an example, I didn't realize we were missing fields from unknown schemas here 👍

**Comment by jakubwasikowski** [comment]
> There is a "Roadmap" section in README - do you think it can be removed now?

I think it can be removed and I did that, but there are two things:
- does that entry in roadmap relate to interpreting property type depending on type of schema?
- I can see that on the page mentioned in README (https://developers.facebook.com/docs/sharing/opengraph/object-properties) there are more schemas than these specified in http://ogp.me/#types, but I have to say that there is a lot of mess and it's quite hard to say what schemas should be defined. Despite that fact, do we want to support some of them?

If there is `yes` answer to any of above questions then we should rather keep that roadmap section.

File: extruct/opengraph.py (hunk 1 above)
Comment (jakubwasikowski) [reply extruct/opengraph.py]: Done

**Comment by kmike** [comment]
@jakubwasikowski good points. Though I think either way README is a wrong place for a roadmap, and supporting all this stuff was not in *our* roadmap anyways, it seems priorities changed.

**Comment by kmike** [comment]
Thanks Jakub!

**Pull request merged**